Support for Mofi 5500

rickmacgillis opened the pull request:
Hi guys. I've created a Mofi 5500 target for OpenWRT. Please merge it in.

rickmacgillis committed:
Add Mofi 500 DTS

Signed-off-by: Rick Mac Gillis <8941225+rickmacgillis@users.noreply.github.com>

rickmacgillis committed:
Support Mofi 5500

Signed-off-by: Rick Mac Gillis <8941225+rickmacgillis@users.noreply.github.com>

rickmacgillis committed:
Mofi 5500 LEDs

Signed-off-by: Rick Mac Gillis <8941225+rickmacgillis@users.noreply.github.com>

mrkiko commented:
Hi!

Thank you for your work and contribution.
I can't find installation / recovery instructions.
I would recommend you to add device support in a single commit and, in case they're missing, add those instructions there or mention their presence in the wiki in case they are present there.

bmork commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -1679,6 +1679,19 @@ define Device/mikrotik_routerboard-m33g
 endef
 TARGET_DEVICES += mikrotik_routerboard-m33g
 
+define Device/mofinetwork_mofi5500-5gxelte
```
Should match the device compatible

bmork commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -1679,6 +1679,19 @@ define Device/mikrotik_routerboard-m33g
+	kmod-wwan kmod-gpio-button-hotplug kmod-hsdma-mtk kmod-mmc kmod-mhi-bus \
+	kmod-mhipci-generic kmod-sdhci kmod-sdhci-mt7620 kmod-usb-core kmod-usb2 \
+	kmod-mac80211 kmod-mt7615e
+  SUPPORTED_DEVICES += mofi5500
```
Won't be necessary when you fix the compatiable and/or device definition

bmork commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -1679,6 +1679,19 @@ define Device/mikrotik_routerboard-m33g
+  DEVICE_PACKAGES := kmod-mt7615-firmware kmod-usb3 kmod-scsi-core kmod-leds-gpio \
+	kmod-wwan kmod-gpio-button-hotplug kmod-hsdma-mtk kmod-mmc kmod-mhi-bus \
+	kmod-mhipci-generic kmod-sdhci kmod-sdhci-mt7620 kmod-usb-core kmod-usb2 \
+	kmod-mac80211 kmod-mt7615e
```
seems excessive.  Are all these extra packages required for basic device operation?  Note that due to the way initramfs images are built shared by the builders, these packages will not necessarily end up in all the "official" images.

rickmacgillis commented:
@mrkiko

> Hi!
> 
> Thank you for your work and contribution. I can't find installation / recovery instructions. I would recommend you to add device support in a single commit and, in case they're missing, add those instructions there or mention their presence in the wiki in case they are present there.

Where do I add the installation instructions? It just gets uploaded to the firmware upgrade page on the router.

@bmork
> Should be "vendor,model"', "mediatek,mt7621-soc"'. See the numerous examples of supported devices.
I'll put it back. Originally I made it match the makefile and kept both of those consistent with what Mofi's custom OpenWRT distro expects it to be so it can be uploaded without forcing the flash.

> This is already defined in mt7621.dtsi. You don't need to, and should not, repeat it.

Where? I did a search on mt7621.dtsi and couldn't find `usb@1e1c0000`. What line? xhci has a couple references to the string `usb`, but that's all. Am I missing something?

> Wondering about this. The USB phy seems to be included with the xhci controller in our mt7621.dtsi. Is this a newer/better/correct definition from mainline? If so, should we update mt7621.dtsi? There's no reason to believe this device is different from any other MT7621 device with USB, is there?

So, I decompiled the dtb file from Mofi's firmware, and I saw that it had a usb-phy. I can test it without this and see if both USB ports work or not.

> 0x700 looks weird when the other phys all have 0x100 register space. Why is that so?

That's how I saw it in Mofi's decompiled DTB.
```
usb-phy@0x1e1d0900 {
			reg = <0x1e1d0900 0x700>;
			#phy-cells = <0x01>;
			clocks = <0x13>;
			clock-names = "ref";
			linux,phandle = <0x15>;
			phandle = <0x15>;
		};
```

> phandles are dynamic values generated when compiling this. You can't define it like that

Correct. I missed that when cleaning up what I copied out of the decompiled DTB.

> Is this a U-Boot environment partition? If so, then you will want to add ubootenv-tools support for it, and drop the read-only attribute. Being able to read and modify the U-Boot environment is usually only a good thing, and never bad.

Updated.

> How will OpenWrt use this partition if it is read-only? You won't be able to install any recovery system there then. Or is that by purpose?

I guess it's fine as long as it's getting replaced with another recovery mechanism. I just didn't want it to leave people without a recovery mechanism. Removed.

> Should match the device compatible

I changed device compatible to the following, so now it's correct.
```
compatible = "mofinetwork,mofi5500-5gxelte", "mediatek,mt7621-soc";
```

> Won't be necessary when you fix the compatiable and/or device definition

It's very necessary. In order to upload the firmware without Mofi's firmware saying it's invalid and not giving you an option to "force" the install, the only way to install it if missing is to use the console. I'm going to leave this in.

> seems excessive. Are all these extra packages required for basic device operation? Note that due to the way initramfs images are built shared by the builders, these packages will not necessarily end up in all the "official" images.

Yes, I agree. The problem is that I don't really know how to test which ones are absolutely necessary. If you know how to create that list properly, I'd like to hear it.

All changes I've made will get pushed when the rest of the issues being discussed are taken care of.

rickmacgillis commented:
Cool. I've removed the usb@ and usb-phy definitions and both work correctly still.

I've also added this comment to the makefile.
```
# Do not remove SUPPORTED_DEVICES! The customized Mofi version of OpenWRT (stock firmware) expects
# to see mofi5500 as the device name. The stock firmware does not allow for forcing an installation.
# Without this line, users cannot upload the new firmware through the stock Mofi firmware.
```

rickmacgillis commented:
Thank you. I had to look up how to do more than just a title. For those who are looking for that info, check out https://stackoverflow.com/questions/16122234/how-to-commit-a-change-with-both-message-and-description-from-the-command-li

I'll add that once I have the kmod stuff sorted. Any tips on getting that right?
